tests/CMakeLists.txt

```diff
@@ -1,33 +1,29 @@
-cmake_minimum_required(VERSION 3.8)
-project(nova-renderer)
-
 set(BUILD_SHARED_LIBS ON)
 
 set(INSTALL_GTEST OFF)
+set(BUILD_GMOCK OFF CACHE BOOL "Build GMOCK shared library" FORCE)
+
 add_subdirectory(3rdparty/googletest)
 include(GoogleTest)
 
-if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
-    set(NOVA_CXX_FLAGS "${NOVA_CXX_FLAGS} -pthread -g -rdynamic")
-endif()
+find_package(Threads REQUIRED)
 
 # End-to-end test
-add_executable(nova-test-end-to-end src/end_to_end_runner.cpp src/general_test_setup.hpp)
-add_dependencies(nova-test-end-to-end nova-renderer)
-target_include_directories(nova-test-end-to-end PRIVATE ${CMAKE_CURRENT_LIST_DIR}/3rdparty/googletest/googletest/include)
+add_executable(nova-test-end-to-end ${CMAKE_CURRENT_SOURCE_DIR}/src/end_to_end_runner.cpp ${CMAKE_CURRENT_SOURCE_DIR}/src/general_test_setup.hpp)
```

Comment:
Why the change to the paths to the test files?

Reply:
Because I was compiling c++ as c and I couldn't figure out why so I was trying everything I could to try to get cmake to figure out it was c++... will fix